src/core/server/preboot/preboot_service.test.ts

```diff
@@ -15,6 +14,10 @@ import { configServiceMock, loggingSystemMock } from '../mocks';
 
 import { PrebootService } from './preboot_service';
 
+function nextTick() {
```

Comment:
Why can't we use `import { nextTick } from '@kbn/test-jest-helpers';`? Maybe you could add a comment explaining why we can't import it here.

Reply:
Good point, the package `@kbn/test-jest-helpers` contains code that does some assumptions and uses globals that are only available on a  _browser-like_ environment (using `jsdom` behind the scenes).

This PR modifies server-side tests to run on _node-like_ environment instead. Thus, importing the package breaks the build. Perhaps we could consider splitting the package in 3 separate packages; something on the lines of:

* @kbn/test-jest-common-helpers
* @kbn/test-jest-node-helpers
* @kbn/test-jest-jsdom-helpers

I'll add a comment in the code.

Reply:
@sebelga Don't we want to discourage the use of `nextTick`? If so then maybe we can add some code comments, a `console.log` inside of it that warns folks away from using it, or rename it `nextTickDoNoUse`.

Reply:
@cjcenizal Yes we do want to avoid those. It has been the source of problem for jest timeout (and thus flakiness) in our tests and we managed to not rely on it.

We took the habit of having on top of all of our test files

```js
beforeAll(() => {
  jest.useFakeTimers();
});

afterAll(() => {
  jest.useRealTimers();
});
```

To make sure we didn't run any `setTimeout` or `setInterval` with internal time. We manually handle the time if/when needed with `jest. advanceTimersByTime() / jest. runAllTimers()`.